### PR TITLE
[CHK-2555] Card details refactoring

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Card details fields refactoring

#### List of Changes

Replace maskedPan with lastFourDigits field
Remove cardHolderName field
Change expiryDate format to `"YYYYAA"`. Update validation expression

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/domain/wallets/details/LastFourDigits.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package it.pagopa.wallet.domain.wallets.details
+
+data class LastFourDigits(val lastFourDigits: String) {
+    init {
+        require(Regex("[0-9]{4}").matchEntire(lastFourDigits) != null) {
```
Comment (pietro-tota) [inline src/main/kotlin/it/pagopa/wallet/domain/wallets/details/LastFourDigits.kt:5]: same as above, Kotlin Regex uses java Pattern whose instances are [thread-safe](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html)

**Review by pietro-tota — DISMISSED** [review]
Well done, just some considerations

**Review by Fernando-Granato — APPROVED** [review]
Good work! 💯

**Review by Fernando-Granato — DISMISSED** [review]
Remember to fix the tests on the deployment pipeline

**Review by giovanniberti — APPROVED** [review]

**Review by pietro-tota — APPROVED** [review]

**Review by Fernando-Granato — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

File: src/main/kotlin/it/pagopa/wallet/domain/wallets/details/LastFourDigits.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package it.pagopa.wallet.domain.wallets.details
+
+data class LastFourDigits(val lastFourDigits: String) {
```
Comment (petretiandrea) [inline src/main/kotlin/it/pagopa/wallet/domain/wallets/details/LastFourDigits.kt:3]: Not related to this PR, but have you ever considered using [Kotlin's inline classes](https://kotlinlang.org/docs/inline-classes.html)? The purpose is really to wrap primitive types to have a language closer to the domain. At the same time it reduces the overhead on the heap for generating new "real" objects.

Comment (ciuffagianluca) [reply src/main/kotlin/it/pagopa/wallet/domain/wallets/details/LastFourDigits.kt:3]: Good point! I will face in technical debt issue!

**Pull request merged**